addon-descriptor.js

```diff
@@ -3,70 +3,32 @@
 
 var request = require('request');
-var RSVP = require('rsvp');
 var ngrok = require('ngrok');
-
-var registerAddOnDescriptor = function(hostRegUrl, descriptorUrl, user, password) {
-
-    // default authentication credentials against Confluence
-    var authCredentials = {
-        'user': user,
-        'pass': password,
-        'sendImmediately': true
-    };
-
-    return new RSVP.Promise(function(resolve, reject) {
-        console.log('Request url for getting upm-token ' + hostRegUrl + '/rest/plugins/1.0/');
-        request.head({
-            uri: hostRegUrl + '/rest/plugins/1.0/',
-            jar: false,
-            auth: authCredentials
-        }, function(err, res) {
-
-            console.log('Result of getting upm-token: ' + res.statusCode);
-
-            if (err || (res && (res.statusCode < 200 || res.statusCode > 299))) {
-                return reject([err, res]);
-            }
-
-            var upmToken = res.headers['upm-token'];
-
-            console.log('Request url for updating add-on descriptor ' + descriptorUrl);
-            request.post({
-                uri: hostRegUrl + '/rest/plugins/1.0/?token=' + upmToken,
-                headers: {'content-type': 'application/vnd.atl.plugins.remote.install+json'},
-                body: JSON.stringify({pluginUri: descriptorUrl}),
-                jar: false,
-                auth: authCredentials
-            }, function(err, res) {
-                console.log('Result of updating add-on descriptor: ' + res.statusCode);
-                if (err || (res && res.statusCode !== 202)) {
-                    return reject([err, res]);
-                }
-                console.log('Successfully updated add-on descriptor');
-                resolve();
-            });
-        });
-
-    });
-};
+var upmInstall = require('upm-install');
 
 var waitUntilServersAreReady = function(hostRegUrl, descriptorUrl, user, password) {
-    console.log("Check if the local server and the cloud instance are available ...");
+    console.log("Check if the local server %s and the cloud instance %s are available ...", descriptorUrl, hostRegUrl);
     setTimeout(function() {
-        request.head({
+        request.get({
             uri: hostRegUrl
         }, function(err, res) {
             if (res !== undefined && res.statusCode === 200) {
-                console.log("Cloud instance is available (" + hostRegUrl + ")");
+                green("Cloud instance is available (" + hostRegUrl + ")");
                 request.head({
                     uri: descriptorUrl
                 }, function(err, res) {
                     if (res !== undefined && res.statusCode === 200) {
-                        console.log("Server instance is available (" + descriptorUrl + ")");
-                        registerAddOnDescriptor(hostRegUrl, descriptorUrl, user, password);
+                        green("Server instance is available (" + descriptorUrl + ")");
+                        upmInstall({
```

Comment:
Not really relevant for now, because this project will probably be deprecated soon anyway, but if I had re-implement this functionality, I would use promises all the way down to simplify the flow. Somehow like this:

```JavaScript
function installAddOn(connectionInfo) {
    return waitForAddOnServer(connectionInfo.descriptorUrl)
        .then(() => upmInstall(connectionInfo));
}
```

But as I said, not relevant for now, just as info :-)

Reply:
I agree on the "deprecated soon" (and the promises). Actually firing up ngrok and installing the connector don't need to be cuddled together